circuits/cpp/src/aztec3/circuits/abis/new_contract_data.hpp

```diff
@@ -95,24 +95,6 @@ template <typename NCT> struct NewContractData {
     }
 };
 
-template <typename NCT> void read(uint8_t const*& it, NewContractData<NCT>& new_contract_data)
-{
-    using serialize::read;
-
-    read(it, new_contract_data.contract_address);
-    read(it, new_contract_data.portal_contract_address.address_);
```

Comment:
Seems like here we're reading/writing directly to a field within this struct, which is not the same as msgpack is doing I guess. Is this change ok?

Reply:
This was never needed I believe, see read(portal_contract_address), confusing though!
```
template <typename B> void read(B& it, address& addr)
{
    using serialize::read;
    fr address_field;
    read(it, address_field);
    addr = address(address_field);
}

=> 
    address(fr const& address)
        : address_(address){};
```